src/util/valid-email.ts

```diff
@@ -0,0 +1,6 @@
+// TypeScript adaptation of https://stackoverflow.com/a/46181/1696044
+const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+
+export default (email?: string) => {
+  return email?.trim().length && emailRegex.test(email)
+}
```

Comment:
We may want to see if there's a standard dependency we can use for this eventually